uberfire-api/src/main/java/org/uberfire/workbench/type/Category.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.uberfire.workbench.type;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum Category {
+
+    OTHERS,
```

Comment:
In the ideal world, Category on uberfire api doesn't know anything about the asset types.

My suggestion is to extract this and somehow dinamycally lookup the categories. One option is to use cdi for this, like is done with SocialAdapter and his child classes.

Reply:
another option would be to transform category in a cdi bean, scan all resource type instances (they are all beans) and dynamically lookup for all category types.

Categories are only used by resource types?